File: algorithms/Analysis/Anagram/anagramSolution2.py
def anagramSolution2(s1, s2):
    alist1 = list(s1)
    alist2 = list(s2)

    alist1.sort()
    alist2.sort()

    pos = 0
    ok = True

    while pos < len(s1) and ok:
        if alist1[pos] == alist2[pos]:
            pos += 1
        else:
            ok = False

    return ok

print(anagramSolution2("abcd", "bcda"))

File: algorithms/Analysis/Anagram/anagramSolution2.js
var string1 = "abcd"
var string2 = "bcda"

function anagramSolution2(s1, s2) {
    s1 = s1.split("").sort().join("")
    s2 = s2.split("").sort().join("")

    var pos = 0;
    var ok = true;

    while (pos < s1.length && ok) {
        if (s1[pos] == s2[pos]) {
            pos += 1
        } else {
            ok = false
        }
    }

    return ok
}

console.log(anagramSolution2(string1, string2))